src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -67,23 +69,23 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
         try {
 
             hookCtx = HookContext.from(key, type, this.getMetadata(),
-                openfeatureApi.getProvider().getMetadata(), ctx, defaultValue);
+                    openfeatureApi.getProvider().getMetadata(), ctx, defaultValue);
 
             mergedHooks = ObjectUtils.merge(provider.getProviderHooks(), flagOptions.getHooks(), clientHooks,
-                openfeatureApi.getApiHooks());
+                    openfeatureApi.getApiHooks());
 
             EvaluationContext ctxFromHook = hookSupport.beforeHooks(type, hookCtx, mergedHooks, hints);
 
-            EvaluationContext invocationCtx = EvaluationContext.merge(ctx, ctxFromHook);
+            EvaluationContext invocationCtx = ctx.merge(ctxFromHook);
 
             // merge of: API.context, client.context, invocation.context
-            EvaluationContext mergedCtx = EvaluationContext.merge(
-                    EvaluationContext.merge(
-                            openfeatureApi.getEvaluationContext(),
-                            this.getEvaluationContext()
-                    ),
-                    invocationCtx
-            );
+            EvaluationContext apiContext = openfeatureApi.getEvaluationContext() != null
+                    ? openfeatureApi.getEvaluationContext()
+                    : new MutableContext();
+            EvaluationContext clientContext = openfeatureApi.getEvaluationContext() != null
+                    ? this.getEvaluationContext()
+                    : new MutableContext();
+            EvaluationContext mergedCtx = apiContext.merge(clientContext.merge(invocationCtx));
```

Comment:
Why did this change?

Reply:
Merge was previously a static method. I don't think it can be any longer, since we have to instantiate a new instance, or return one of the existing ones (so we need a `this`).

Previously, `merge` could easily handle if once of the context passed were null. With the new instance method, we have to make sure the the context is not null before calling it. In many of the tests, `this.getEvaluationContext()` and `openfeatureApi.getEvaluationContext()` were null.